scipy/stats/morestats.py

```diff
@@ -2721,50 +2722,109 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False):
         The second set of measurements.  If `y` is not given, then the `x`
         array is considered to be the differences between the two sets of
         measurements.
-    zero_method : string, {"pratt", "wilcox", "zsplit"}, optional
+    zero_method : {"pratt", "wilcox", "zsplit"}, optional
         "pratt":
-            Pratt treatment: includes zero-differences in the ranking process
-            (more conservative)
+            includes zero-differences in the ranking process,
+            but drop ranks of the zeros, see [4]_, (more conservative)
         "wilcox":
-            Wilcox treatment: discards all zero-differences
+            Wilcox treatment: discards all zero-differences, the default
         "zsplit":
             Zero rank split: just like Pratt, but splitting the zero rank
             between positive and negative ones
     correction : bool, optional
         If True, apply continuity correction by adjusting the Wilcoxon rank
         statistic by 0.5 towards the mean value when computing the
         z-statistic.  Default is False.
+    alternative : {"two-sided", "greater", "less"}, optional
+        The alternative hypothesis to be tested, see Notes. Default is
+        "two-sided".
 
     Returns
     -------
     statistic : float
-        The sum of the ranks of the differences above or below zero, whichever
-        is smaller.
+        If `alternative` is two-sided, the sum of the ranks of the
+        differences above or below zero, whichever is smaller.
+        Otherwise the sum of the ranks of the differences above zero.
     pvalue : float
-        The two-sided p-value for the test.
+        The p-value for the test depending on `alternative`.
+
+    See Also
+    --------
+    kruskal, mannwhitneyu
 
     Notes
     -----
-    Because the normal approximation is used for the calculations, the
-    samples used should be large.  A typical rule is to require that
-    n > 20.
+    The test has been introduced in [4]_. Given n independent samples
+    (xi, yi) from a bivariate distribution, compute the differences
+    di = xi - yi. One assumption of the test is that the differences
+    are symmetric, see [2]_.
+    The two-sided test has the null hypothesis that the median of the
+    differences is zero against the alternative that it is different from
+    zero. The one-sided test has the null that the median is positive against
+    the alternative that the it is negative (``alternative == 'less'``),
+    or vice versa (``alternative == 'greater.'``).
+
+    The test uses a normal approximation to derive the p-value. A typical rule
+    is to require that n > 20. For smaller n, exact tables can be used to find
+    critical values.
 
     References
     ----------
     .. [1] https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test
+    .. [2] Conover, W.J., Practical Nonparametric Statistics, 1971.
+    .. [3] Pratt, J.W., Remarks on Zeros and Ties in the Wilcoxon Signed
```

Comment:
You can also add 10.1080/01621459.1959.10501526 for Pratt and 10.2307/3001968 for Wilcoxon. I think we even have a rubric for DOI for automatic linking.

Reply:
I added the whole URL, is that the right way?

Reply:
I believe that should be ```:doi:`10.1080/01621459.1959.10501526` ``` instead

Reply:
yes, that is used in `moyal` in stats and works. i changed it